java/arcs/sdk/android/storage/service/StorageService.kt

```diff
@@ -123,6 +124,28 @@ class StorageService : ResurrectorService() {
             """.trimMargin("|")
         )
 
+        // Dump current process global binder stats to understand contention in the thread pool,
+        // peak usage of shared binder memory, the number of pending transactions, etc.
+        // Hide the dump when failing to fetch the stats from the kernel binder driver.
+        mapOf(
+            "Memory high watermark (pages)" to "pages high watermark",
+            "Pending transactions" to "pending transactions",
+            "Requested threads" to "requested threads",
+            "Ready threads" to "ready threads"
+        ).run {
+            AndroidBinderStats.query(*values.toTypedArray()).iterator().let { stats ->
+                mapValues { stats.next() }
+            }
+        }.run { if (any { (_, v) -> v.isNotEmpty() }) this else null
+        }?.run {
```

Comment:
```suggestion
        }.takeIf { any { (_, v) -> v.isNotEmpty() } }?.run {
```

Reply:
Better! I didn't know the takeIf(...), thank you.